samples/net/openthread/coprocessor/src/main.c

```diff
@@ -12,39 +12,50 @@ LOG_MODULE_REGISTER(ot_br, LOG_LEVEL_DBG);
 #include <zephyr.h>
 #include <usb/usb_device.h>
 #include <drivers/uart.h>
+#include <net/openthread.h>
+#include <openthread/ncp.h>
 
 #define APP_BANNER "***** OpenThread NCP on Zephyr %s *****"
 
 void main(void)
 {
-#if defined(CONFIG_OPENTHREAD_COPROCESSOR_SPINEL_ON_UART_ACM)
-	const struct device *dev;
-	uint32_t baudrate = 0U;
-	int ret;
-
-	dev = device_get_binding(
-		CONFIG_OPENTHREAD_COPROCESSOR_SPINEL_ON_UART_DEV_NAME);
-	if (!dev) {
-		LOG_ERR("UART device not found");
-		return;
+	if (IS_ENABLED(CONFIG_OPENTHREAD_COPROCESSOR_SPINEL_ON_UART_ACM)) {
+		int ret;
+		const struct device *uart_dev;
+		uint32_t dtr = 0U;
+
+		ret = usb_enable(NULL);
```

Comment:
Just wondering here... this looks like a part of the USB-CDC-specific UART module initialization code, perhaps it would make sense to move it inside `otPlatUartEnable()`? It'd be consistent with the vanilla OT NCP app, where it doesn't bother with low-level platform initialization. I believe the NCP initialization part could then be left as it is and the actual app would be even simpler (so less code to duplicate in case NCP module is a part of a larger app). What do you think?

Reply:
Frankly speaking the `usb_enable()` used to be called from `otPlatUartEnable()` but it has been changed by this commit: c6f7d87142739036fdae6dbc722f97ec9acef4df. I agree that calling `usb_enable()` from `otPlatUartEnable()` is cleaner solution.

Reply:
Ok, thank for claryfing. So it seems to be USB limitation then, I guess we have to live with that for now...

Reply:
I needed to change the fix for `Framing error`. Previous proposal broke UART communication with `wpantund`.
Eventually I placed `usb_enable()` in `otPlatUartEnable()` and added there the fix.  
The issue, which worried me (#27427 and #27071) , concerned only USB UART console implementation which co-processor sample does not use.